light/lightchain.go

```diff
@@ -406,11 +406,11 @@ func (lc *LightChain) InsertHeaderChain(chain []*types.Header, checkFreq int, co
 		status, err := lc.hc.WriteHeader(header)
 
 		switch status {
-		case core.CanonStatTy:
+		case types.CanonStatTy:
```

Comment:
Why the move from core to types? This change will probably introduce quite a few extra conflicts when merging from upstream, and since the status is only used in the `core` and `light` packages, moving them to types doesn't seem to add any value.

Reply:
Because `WriterStatus` is introduced into `consensus` package, here:
https://github.com/celo-org/celo-blockchain/blob/f3d394618ceeed33d54fbd8140223c1c31bfc086/consensus/consensus.go#L185
it causes import cycle if not moved to types.
An alternative way is to remove `WriteStatus`, only return error. Since we don't use `WriteStatus` in worker anyway.
wdyt?

Reply:
Yeah that seems like a good approach, It looks like the status from the  public method `Blockchain.WriteBlockWithState` is never used, its only used internally inside `Blockchain`.

Reply:
+1 to keeping `types.WriteStatus` in blockchain, and not exposing it as a result of `WriteBlockWithState`